### PR TITLE
Simplex Perlin3D NME Block

### PR DESCRIPTION
**Pull request by Pryme8**
Let me know if you see anything that needs to be changed.

Also it says I am 7 commits behind the master, I assume there are no conflicts though.

### CODE REVIEW

File: nodeEditor/src/blockTools.ts [hunk 1]
```diff
@@ -54,10 +54,13 @@ import { ReciprocalBlock } from 'babylonjs/Materials/Node/Blocks/reciprocalBlock
 import { GradientBlock } from 'babylonjs/Materials/Node/Blocks/gradientBlock';
 import { WaveBlock, WaveBlockKind } from 'babylonjs/Materials/Node/Blocks/waveBlock';
 import { NodeMaterial } from 'babylonjs/Materials/Node/nodeMaterial';
+import { SimplexPerlin3DBlock } from 'babylonjs/Materials/Node/Blocks/simplexPerlin3DBlock';
 
 export class BlockTools {
     public static GetBlockFromString(data: string, scene: Scene, nodeMaterial: NodeMaterial) {
         switch (data) {
+			case "SimplexPerlin3DBlock":
```
Comment (deltakosh) [inline nodeEditor/src/blockTools.ts]: indentation issue

File: nodeEditor/src/components/nodeList/nodeListComponent.tsx [hunk 2]
```diff
@@ -33,7 +33,8 @@ export class NodeListComponent extends React.Component<INodeListComponentProps,
             Interpolation: ["LerpBlock", "SmoothStepBlock", "NLerpBlock"],
             Matrices: ["Matrix", "WorldMatrixBlock", "WorldViewMatrixBlock", "WorldViewProjectionMatrixBlock", "ViewMatrixBlock", "ViewProjectionMatrixBlock", "ProjectionMatrixBlock"],
             Mesh_Attributes: ["InstancesBlock", "PositionBlock", "UVBlock", "ColorBlock", "NormalBlock", "TangentBlock", "MatrixIndicesBlock", "MatrixWeightsBlock", "WorldPositionBlock", "WorldNormalBlock", "FrontFacingBlock"], 
-            Output_Blocks: ["VertexOutputBlock", "FragmentOutputBlock", "DiscardBlock"],
+            Noises: ["SimplexPerlin3DBlock"],
+			Output_Blocks: ["VertexOutputBlock", "FragmentOutputBlock", "DiscardBlock"],
```
Comment (deltakosh) [inline nodeEditor/src/components/nodeList/nodeListComponent.tsx]: indent

File: src/Materials/Node/Blocks/simplexPerlin3DBlock.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        if (!this._outputs[0].hasEndpoints) {
+            return;
+        }
+        
+        let functionString = `const float SKEWFACTOR = 1.0/3.0;\r\n`;
+        functionString += `const float UNSKEWFACTOR = 1.0/6.0;\r\n`;
+        functionString += `const float SIMPLEX_CORNER_POS = 0.5;\r\n`;        
+        functionString += `const float SIMPLEX_TETRAHADRON_HEIGHT = 0.70710678118654752440084436210485;\r\n`;
+        functionString += `float SimplexPerlin3D( vec3 P ){\r\n`;
+        functionString += `P *= SIMPLEX_TETRAHADRON_HEIGHT;\r\n`;
```
Comment (deltakosh) [inline src/Materials/Node/Blocks/simplexPerlin3DBlock.ts]: please add some tabs to make the generated code cleaner to read :)

Comment (Pryme8) [reply src/Materials/Node/Blocks/simplexPerlin3DBlock.ts]: Yes sir and what is the quickest way to do the linting without a fill rebuild?

File: nodeEditor/src/components/nodeList/nodeListComponent.tsx (hunk 2 above)
Comment (Pryme8) [reply nodeEditor/src/components/nodeList/nodeListComponent.tsx]: ahhh dumb dumb dumb, I forgot my notpad++ had tabs set to tabs not spaces on my laptop....

**Comment by Pryme8** [comment]
What this about:
This branch has conflicts that must be resolved
Only those with write access to this repository can merge pull requests.
Conflicting files
nodeEditor/src/components/nodeList/nodeListComponent.tsx

**Comment by deltakosh** [comment]
You need to learn git my friend :)

**Comment by deltakosh** [comment]
Waiting for the CI to finish

**Comment by Pryme8** [comment]
Let me do a build, looks like the linting did not get fixed.

File: src/Materials/Node/Blocks/simplexPerlin3DBlock.ts (hunk 3 above)
Comment (deltakosh) [reply src/Materials/Node/Blocks/simplexPerlin3DBlock.ts]: gulp lint or something (asking @sebavan)

Comment (Pryme8) [reply src/Materials/Node/Blocks/simplexPerlin3DBlock.ts]: gulp cleanup && gulp fullLint

Comment (Pryme8) [reply src/Materials/Node/Blocks/simplexPerlin3DBlock.ts]: actually looks like there are a bunch you have to run... damn maybe ill just stick with the build

**Pull request merged**